scripts/process-reboot-cause

```diff
@@ -69,6 +71,84 @@ def read_reboot_cause_files_and_save_state_db():
+        if not platform:
+            raise KeyError("Platform key missing from platform_info")
+
+        # Path to platform.json
+        platform_path = os.path.join("/usr/share/sonic/device", platform, "platform.json")
+        sonic_logger.log_info(f"Reading platform JSON from: {platform_path}")
+
+        with open(platform_path, 'r') as f:
+            platform_data = json.load(f)
+
+        dpus = platform_data.get('DPUS', [])
+        if not dpus:
+            sonic_logger.log_warning("No DPUS found in platform.json")
+
+    except Exception as e:
+        sonic_logger.log_error(f"Error retrieving DPUs: {e}")
+
+    return dpus
+
+def get_sorted_reboot_cause_files(dpu_history_path):
+    """Retrieve and sort the reboot cause files for a specific DPU."""
+    try:
+        # Assuming you have a way to list the files in the directory
+        files = os.listdir(dpu_history_path)
+        # Filter and sort the files based on your criteria (e.g., by modification time)
+        sorted_files = sorted(
+            [os.path.join(dpu_history_path, f) for f in files if f.endswith('.json')],
+            key=os.path.getmtime,  # Sort by modification time
+            reverse=True  # Most recent first
+        )
+        return sorted_files
+    except Exception as e:
+        sonic_logger.log_error(f"Error retrieving reboot cause files for {dpu_history_path}: {e}")
+        return []
+
+
+def read_dpu_reboot_cause_files_and_save_chassis_state_db():
+    """Retrieve reboot cause from history files and save them to StateDB."""
+    try:
+        # Get the DPUs from the platform configuration
+        dpus = get_dpus()
+        history_dir = '/host/reboot-cause/module'
+
+        for dpu in dpus:
+            dpu_history_path = os.path.join(history_dir, dpu)
+
+            # Get sorted reboot cause files for the DPU
+            reboot_files = get_sorted_reboot_cause_files(os.path.join(dpu_history_path, "history"))
+
+            for reboot_file in reboot_files:
+                if os.path.isfile(reboot_file):
+                    with open(reboot_file, "r") as cause_file:
+                        try:
+                            data = json.load(cause_file)
+                            # Ensure keys exist
+                            if 'gen_time' not in data:
+                                sonic_logger.log_warning(f"Missing 'gen_time' in data from {reboot_file}")
+                                continue  # Skip this file
+
+                            _hash = f"{REBOOT_CAUSE_TABLE_NAME}|{data['gen_time']}"
+                            state_db.set(state_db.STATE_DB, _hash, 'cause', data.get('cause', ''))
+                            state_db.set(state_db.STATE_DB, _hash, 'time', data.get('time', ''))
+                            state_db.set(state_db.STATE_DB, _hash, 'user', data.get('user', ''))
+                            state_db.set(state_db.STATE_DB, _hash, 'comment', data.get('comment', ''))
```

Comment:
@rameshraghupathy function says chassis STATE_DB but here its STATE_DB?

Reply:
@prgeor  Fixing the function name